components/discover/ErrorModal.tsx

```
@@ -0,0 +1,27 @@
+  return (
+    <div className="fixed inset-0 bg-black bg-opacity-50 flex justify-center items-center z-50">
+      <div className="bg-background rounded-lg p-6 w-3/4 max-w-sm">
+        <h2 className="text-center text-xl font-bold text-red-500">Failed</h2>
+        <p className="text-center text-muted-foreground mt-2">
+          Failed to load listings feed.
+        </p>
+        <div className="my-4 border-b"></div>
+        <button
+          onClick={onRetry}
+          className="w-full bg-primary text-white py-2 rounded-lg hover:bg-primary/90"
+        >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)